src/refract/VisitorUtils.h

```diff
@@ -119,6 +119,41 @@ namespace refract
         }
     };
 
+    template <typename T>
+    void CheckMixinParent(refract::IElement* element)
+    {
+        const T* resolved = TypeQueryVisitor::as<T>(element);
+
+        if (!resolved) {
+            throw snowcrash::Error("mixin base type should be the same as parent base type. objects should contain object mixins. arrays should contain array mixins", snowcrash::MSONError);
```

Comment:
Why are you throwing `snowcrash::Error` and not some `refract` error? We are in `refract` namespace